From 6a3031aab9e7db59ab4d5ed0f3ce0e484c8f3d39 Mon Sep 17 00:00:00 2001
From: Thomas Tai <thomas.tai@windriver.com>
Date: Thu, 8 Apr 2010 21:49:48 -0400
Subject: [PATCH] change PVR to match 8641D MILS hypervisor

Add 8641 pvr value for MILS hypervisor, also added #error if
user selected unsupported CPU.

Signed-off-by: Thomas Tai <thomas.tai@windriver.com>
---
 arch/powerpc/kernel/vbi/wrhv.c |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/kernel/vbi/wrhv.c b/arch/powerpc/kernel/vbi/wrhv.c
index 08634b7..cdb9669 100644
--- a/arch/powerpc/kernel/vbi/wrhv.c
+++ b/arch/powerpc/kernel/vbi/wrhv.c
@@ -578,12 +578,15 @@ unsigned int wrhv_irq_of_parse_and_map(struct device_node *dev, int index)
 
 unsigned int wrhv_get_pvr(void)
 {
+#if defined(CONFIG_E500)
        return 0x80200000;
+#elif defined(CONFIG_6xx)
+       return 0x80040000;
+#else
+#error unsupported CPU family
+#endif
 }
 
-
-
-
 /* arch/powerpc/kernel/time.c */
 #define HWTIMER_USE_JIFFY 1
 static void wrhv_set_mode(enum clock_event_mode mode,
-- 
1.6.5.2

